File: src/tests/fidl/source_compatibility/enum-flexible-strict/dart/step_01_after.dart
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:fidl_fidl_test_enumflexiblestrict/fidl_async.dart' as fidllib;

// [START contents]
fidllib.Color complement(fidllib.Color color) {
  assert(color.isUnknown() == false);
  switch (color) {
    case fidllib.Color.blue:
      return fidllib.Color.red;
    case fidllib.Color.red:
      return fidllib.Color.blue;
    default:
      return null;
  }
}

// [END contents]

File: src/tests/fidl/source_compatibility/enum-flexible-strict/hlcpp/step_01_after.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/test/enumflexiblestrict/cpp/fidl.h>  // nogncheck
namespace fidl_test = fidl::test::enumflexiblestrict;

// [START contents]
fidl_test::Color complement(fidl_test::Color color) {
  switch (color) {
    case fidl_test::Color::RED:
      return fidl_test::Color::BLUE;
    case fidl_test::Color::BLUE:
      return fidl_test::Color::RED;
    default:
      return color;
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }
